### PR TITLE
Qsub fix

### PR DESCRIPTION
**Pull request by charles-cowart**
Added exit_status parsing for jobs
Added exit_status parsing for non-array jobs as they appear to exit with 'C' job-status, even when an error has occured.
ConvertJob() will now catch any PipelineErrors raised by run() and re-raise them as a ConvertJob-specific Error.
This error should help detect when bcl-convert has exited with 1.

### CODE REVIEW

**Comment by charles-cowart** [comment]
Note that unittests will provide little coverage of the affected code.

File: sequence_processing_pipeline/ConvertJob.py [hunk 1]
```diff
@@ -94,8 +94,8 @@ def _generate_job_script(self):
 
         lines.append(f"#PBS -l pmem={self.pmem}")
 
-        lines.append(f"#PBS -o localhost:{self.log_path}/qsub_stdout.log")
-        lines.append(f"#PBS -e localhost:{self.log_path}/qsub_stderr.log")
+        # lines.append(f"#PBS -o localhost:{self.log_path}/qsub_stdout.log")
```
Comment (antgonza) [inline sequence_processing_pipeline/ConvertJob.py:97]: rm

File: sequence_processing_pipeline/ConvertJob.py [hunk 2]
```diff
@@ -149,9 +149,7 @@ def run(self):
         Run BCL2Fastq/BCLConvert conversion
         :return: None
         '''
-        try:
-            job_info = self.qsub(self.job_script_path, None, None)
-        except PipelineError as e:
-            raise PipelineError(f"ConvertJob failed: {str(e)}")
+        job_info = self.qsub(self.job_script_path, None, None,
```
Comment (antgonza) [inline sequence_processing_pipeline/ConvertJob.py:152]: can you make sure that all calls to self.qsub are outside a try/except?

File: sequence_processing_pipeline/FastQCJob.py [hunk 3]
```diff
@@ -205,8 +205,8 @@ def _generate_job_script(self):
         lines.append(f"#PBS -l mem={self.jmem}")
 
         log_path = join(self.log_path, job_name + '_${PBS_ARRAYID}_')
-        lines.append(f"#PBS -o localhost:{log_path}_stdout.log")
-        lines.append(f"#PBS -e localhost:{log_path}_stderr.log")
+        # lines.append(f"#PBS -o localhost:{log_path}_stdout.log")
```
Comment (antgonza) [inline sequence_processing_pipeline/FastQCJob.py:208]: rm

File: sequence_processing_pipeline/QCJob.py [hunk 4]
```diff
@@ -220,8 +220,8 @@ def _generate_job_script(self, project_name, adapter_a, adapter_A,
         lines.append(f"#PBS -l mem={self.jmem}")
 
         log_path = join(self.log_path, job_name + '_${PBS_ARRAYID}_')
-        lines.append(f"#PBS -o localhost:{log_path}_stdout.log")
-        lines.append(f"#PBS -e localhost:{log_path}_stderr.log")
+        # lines.append(f"#PBS -o localhost:{log_path}_stdout.log")
```
Comment (antgonza) [inline sequence_processing_pipeline/QCJob.py:223]: rm

**Review by antgonza — CHANGES_REQUESTED** [review]

**Comment by charles-cowart** [comment]
@antgonza Hi Antonio, can you please review this one more time? As you can see, there are no cases where qsub is being wrapped in a try/except in this codebase.

**Pull request merged**